lib/bundlex/toolchain/gcc.ex

```diff
@@ -7,11 +7,11 @@ defmodule Bundlex.Toolchain.GCC do
   @compilers %Compilers{c: "gcc", cpp: "g++"}
 
   @impl Toolchain
-  def compiler_commands(native, native_interface \\ nil) do
+  def compiler_commands(native, native_interface) do
     {cflags, lflags} =
       cond do
         native.type == :native && native_interface == :nif -> {"-fPIC", "-rdynamic -shared"}
-        native.type == :lib -> {"-fPIC", ""}
+        native_interface == :lib -> {"-fPIC", ""}
```

Comment:
`native_interface` can't be `:lib`. It can be `:nif`, `:cnode` or `:port`. 
`native.type` can be `:native` or `:lib`. This was the reason why I didn't remove `type` from `native` struct i.e. we still have to know what is the interface of a lib